tdrs-backend/tdpservice/parsers/validators/category1.py

```diff
@@ -0,0 +1,215 @@
+"""Overloads and custom validators for category 1 (preparsing)."""
+
+from tdpservice.parsers.models import ParserErrorCategoryChoices
+from tdpservice.parsers.util import fiscal_to_calendar, year_month_to_year_quarter
+from .base import ValidatorFunctions
+from .util import ValidationErrorArgs, make_validator, _is_all_zeros, _is_empty, value_is_empty
+
+
+def format_error_context(eargs: ValidationErrorArgs):
+    """Format the error message for consistency across cat1 validators."""
+    return f'{eargs.row_schema.record_type} Item {eargs.item_num} ({eargs.friendly_name}):'
+
+
+class PreparsingValidators():
+    """Overloaded base and custom validators for preparsing."""
+
+    @staticmethod
+    def recordIsNotEmpty(start=0, end=None, **kwargs):
+        """Return a function that tests that a record/line is not empty."""
+        return make_validator(
```

Comment:
Using make validator from util function is confusing. This is a good candidate for using Decorator called @validator or @make_validator.

Reply:
i made this change. fwiw i left `make_validator` as a function that gets used in a few places where it wasn't as easy to swap in the decorator version. it's still possible, just a little bit more reorganizing. tell me what you think of what i've done so far and i can keep going if we think it's worthwhile